### ISSUE TITLE
Button is moving in Notification:

### ISSUE THREAD

**Issue by naamagr** [wix/wix-style-react#1243]
### Summary:

Given certain conditions (listed below), the action button in the notification in business manager is animated, and moves back and forth horizontally.

### Priority: 

Blocker

### Steps to reproduce:

To see the notification:
1. Create a site that's not a store (for example blog)
2. Go to dashboard (wix.com)
3. Click on 'add app'
4. Add 'wix stores'
5. Don't! go to editor
6. Go to dashboard again
7. Click on 'wix stores'
8.turn on FT specs.wos.ShouldShowHybridAppToast

To make the button start moving:
1. Go to ‘settings’ on the sidebar
2. Go back to ‘Main Menu’ from the sidebar
3. Click on ‘Orders’ in the side bar

**Comment by argshook** [comment]
Thanks for reporting!

**tl;dr;** css module names are clashing, who could've thought.

---

this is really interesting. It's not issue with wix-style-react itself but rather how css dependencies are handled within business manager.

Here's how it goes:

1. you go to settings, [`settings-app.min.css`](https://static.parastorage.com/services/business-settings/1.172.0/settings-app.min.css) is loaded
1. that file has `._21B-4` classname with "bouncy" animation. This animation comes from tooltip and is fine by itself.
1. you go to "Store Orders", more css is loaded but previously loaded `settings-app.min.css` with `._21B-4` remains
1. you are now in "Store Orders" page, there is notification with button. Guess what classname that button has. bingo, it's `._21B-4`

so, again, tl;dr; is css modules name clash.

We can't really do much in wix-style-react.
Well, we could play with classnames or maybe try and remove that classname from `Notification.ActionButton`. But it wont solve root cause and we're still left with potential future (or existing but unknown) clashes.

What we can do is:
1. talk to business manager guys to see if it's feasible to remove css dependencies after navigating out of a page (probably not, because caching)
1. talk to haste guys to try and adjust css modules pattern to be more random
1. ???

@lbelinsk

**Comment by shlomitc** [comment]
+ @ronami

**Comment by ranyitz** [comment]
@argshook @ronami @lbelinsk @shlomitc 

Hey, everyone.
After looking into this problem I can verify that Arijus is right, css modules names indeed clash.
@ronami and I reproduced it locally using the `css-modules` hashing algorithm and the different props of the matching classes.

## How does the css modules algorithm work, and what are the odds?
The hashing algorithm takes into account 4 things:
1. `hashPrefix` - The name of the project (taken from `package.json`, for example `business-manager`)
2. Relative path to the file -  For example `../node_modules/wix-style-react/dist/src/Backoffice/Tooltip/Tooltip.scss`
3. `localName` - The name of the class, for example `bounce-on-right`
4. `number of digits` - The number of digits that each class is constructed from.

Today we have a hash that built from `5` `base64` digits, that means `1073741824` possibilities of class names.
The clash happens between `business manager` and `business setting` bundles using `wsr` components.

**note:** The possibility of a clash is raising when using `wsr` and different prefixes, because we are generating another X classes for each `wsr` usage in a project.
## Optional solutions

1. Add another digit to the hashing algorithm.
> Will solve this problem, but will increase bundle size of every haste user in approximately `numberOfClasses`*`2`
2. Add another digit only to the business manager bundle as their bundle remains while every other change.
> There are other use cases when multiple projects that using `wsr` are being loaded together.
3. Change one of the class names.

* About "talk to business manager guys to see if it's feasible to remove css dependencies after navigating out of a page", It's not only feasible, but they **have to do it** when using old stack projects (with global class definitions), so they actually do it by default. This is not the case here, because the clash is with the business manager itself.

## So, what should we do?
IMHO The best option we have here is to change the class name of the `bounce-on-right` class to something else. It's not optimal, I know, but the other solutions will affect performance, and that's a fact.

This is not about solving the root cause of the problem, it's about probability and about making the right tradeoffs.
The chance of this to happen will be small if we'll increase the hash length, but it is also possible. For example, if we had a 6 digits hash and we would have encountered the same problem?
If we'll see this problem coming back in other forms, we'll be smarter and we'll know what to do.

What do you say?

**Comment by shlomitc** [comment]
@ranyitz - Thanks for the thorough investigation!
I'm completely OK with fixing the current class name if it can solve the issue quickly without affecting every other project.

**Comment by lbelinsk** [comment]
Closing this issue, but we might need to consider adding this to the backlog (the end solution)
